app/src/main/java/ch/epfl/qedit/model/answer/MatrixModel.java

```diff
@@ -29,4 +35,22 @@ public String getAnswer(int row, int col) throws IndexOutOfBoundsException {
             throw new IndexOutOfBoundsException();
         }
     }
+
+    @Override
+    public boolean equals(@Nullable Object o) {
+        if (o instanceof MatrixModel) {
```

Comment:
If you want to reduce the indentation hell, we can change it to `if (!(o instanceof MatrixModel)) return false;`. For the rest, though, I have no idea how we could reduce the nesting...

Reply:
I do not know either, maybe you could define a method for the inner most `if`but that's kind of artificial.

Reply:
I will try something with a boolean that is true at first and then an "&=" operator for each loop iteration.